### PR TITLE
Prosody security

### PR DESCRIPTION
**Pull request by andir**
###### Motivation for this change

See commit messages.

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages

### CODE REVIEW

**Comment by ajs124** [comment]
Should we backport both commits? The tests seems broken on 21.11 as well.

Also reading their announcement, our jitsi module is affected by this, since I added websocket support to it in https://github.com/NixOS/nixpkgs/pull/133141 and enabled it by default -.-

**Review by picnoir — APPROVED** [review]
LGTM

**Comment by NinjaTrappeur** [comment]
Yes, I think backporting this makes sense.

File: nixos/tests/all-tests.nix [hunk 1]
```diff
@@ -397,7 +397,6 @@ in
   prometheus = handleTest ./prometheus.nix {};
   prometheus-exporters = handleTest ./prometheus-exporters.nix {};
   prosody = handleTest ./xmpp/prosody.nix {};
-  prosodyMysql = handleTest ./xmpp/prosody-mysql.nix {};
```
Comment (zowoq) [inline nixos/tests/all-tests.nix:400]: Needs to be removed here as well.

https://github.com/NixOS/nixpkgs/blob/b47203b28f5cf1efc4e3476ca8f333db3a2c3223/pkgs/servers/xmpp/prosody/default.nix#L89

Comment (NinjaTrappeur) [reply nixos/tests/all-tests.nix:400]: Crap... Thanks for the heads up.

Comment (NinjaTrappeur) [reply nixos/tests/all-tests.nix:400]: https://github.com/NixOS/nixpkgs/pull/155126

**Pull request merged**